client/index.js

```diff
@@ -219,6 +219,15 @@ function reloadApp() {
     }
   } else {
     log.info('[WDS] App updated. Reloading...');
-    self.location.reload();
+    let rootWindow = self;
+    // use parent window for reload (in case we're in an iframe with no valid src)
+    do {
```

Comment:
this is almost never a good idea in the browser - on the off chance that the `while` condition is not met, this will lock the browser forcing a close. please use `setInterval` in conjunction with `clearInterval`

Reply:
I don't really get what you mean with using `setInterval` and `clearInterval` here and how this will improve the code.

+1 for moving the log down